db/migrate/20180829205032_recovery_codes_add_column_and_indices.rb

```diff
@@ -0,0 +1,7 @@
+class RecoveryCodesAddColumnAndIndices < ActiveRecord::Migration[5.1]
+  disable_ddl_transaction!
+  def change
+    add_index :recovery_codes, :user_id, algorithm: :concurrently
+    add_column :recovery_codes, :when_user, :timestamp
```

Comment:
`when_used`?

Might be better to keep it in line with many of the other timestamps and name it `used_at`.